lib/ecto/embedded.ex

```diff
@@ -111,6 +113,19 @@ defmodule Ecto.Embedded do
     |> apply_embeds(struct)
   end
 
+  defp run_prepare(changeset, repo) do
```

Comment:
this is copy-paste from: https://github.com/wojtekmach/ecto/blob/e8fbe9ad269c0cb5d7c7f544249e537ba930f9a5/lib/ecto/repo/schema.ex#L481

In Repo.Schema.run_prepare we get `prepare` list as separate argument because we use that argument in `wrap_in_transaction` call as well. Also, in Repo.Schema changeset got repo filled in using `put_repo_and_action`.

I wasn't sure if I should make that code more similar, so I tried making the minimal change only.

Reply:
@wojtekmach we could move the `Ecto.Changeset.Relation.run_prepare(changeset, prepare)` but I don't think the gain is clear, I can see both sides of the argument. Therefore I will gladly leave this up to you. :) Your call!

Reply:
Let's leave this as is for now. I'm gonna read whole embedded.ex and relation.ex one of these days :)